liquibase/liquibase
# fix: Use default localhost finder for MacOs - fix #2098

This will fix #2098

The NetUtil retrieves a 'wrong' localhost addr on local, that is impossible to map under 30 seconds, using the standard way makes it instant.

- - -
## Dev Handoff Notes (Internal Use)
#### Links
* Fixed Issue: https://github.com/liquibase/liquibase/issues/2098
* Test Results: https://github.com/liquibase/liquibase/pull/2134/checks

#### Testing
* Steps to Reproduce: https://github.com/liquibase/liquibase/issues/2098
* Guidance:
  * Impact: Changes hostname lookup used in a variety of places. Easiest to see may be in the databasechangeloglock table
  * Change unifies logic across linux/mac/windows, but the underlying java calls may act differently in different OS's
  * To ensure that the hostname is working correctly:
    *  run `liquibase update`
    * ctrl-c to kill liquibase after it has locked the database
    *  `liquibase list-locks` to ensure that the hostname is not localhost
  * To test speed, run a `liquibase update` with an empty changelog so that there is little to do. On mac, it should be about as fast as windows/linux

#### Dev Verification
Ran updated NetUtilTest on linux and windows. Don't have a mac to run it on

#### Test Requirements (Liquibase Internal QA)
Performance testing is out of scope for this issue. QA will validate that the correct hostname is used and that update is successful, but not how long it takes an update to run.

 * **Setup** 
-  You will need a Mac and a Windows machine to test this fix.
- The database platform does not matter but example snippets are for MySQL.
- On the database, ensure AUTOCOMMIT is disabled.
- Create a table, "accounts":

```
CREATE TABLE accounts (
    name VARCHAR(100) NOT NULL,
    balance DEC(15,2) NOT NULL
);
```

- In a database IDE, start an insert transaction to the accounts table:
```
BEGIN;

INSERT INTO accounts(name,balance)
VALUES('Abernathy',50000);
```

- Do not commit the insert!!
- Create a formatted SQL file named addColumn-changelog.sql with the following content:
```
--liquibase formatted sql

--changeset liquibase:lb2180
ALTER TABLE accounts
ADD COLUMN phone VARCHAR(15);`
```

**Manual Tests**

**TEST 1: Verify the DATABASECHANGELOGLOCK.LOCKEDBY has the test machine's name and _not_ localhost.**
 Start a Liquibase update to add a column to table accounts: 
 * `liquibase update --changelog-file addColumn-changelog.sql`
 
 In another command prompt, run:
*  `liquibase list-locks`
The list-locks command outputs the lock information with the machine name, for example:
 ` - LENOVO (172.21.224.1) at Dec 2, 2021 3:54:40 PM` , where LENOVO is the machine name.

**TEST 2: Verify an update operation is successful.**
Execute Liquibase update:
*  `liquibase update --changelog-file addColumn-changelog.sql`


* Automated Tests
  * No new functional or harness tests required for this fix.

┆Issue is synchronized with this [Jira Bug](https://datical.atlassian.net/browse/LB-2180) by [Unito](https://www.unito.io)
